src/analysis/normed_space/conformal.lean

```diff
@@ -0,0 +1,211 @@
+real differentiable and its differential at that point is a nonzero constant multiple of
+a linear isometry equivalence.
+
+## Main results
+
+* `conformal_at`: the main definition
+* The conformality of the composition of two conformal maps, the identity map
+  and multiplications by a constant
+* `conformal_at_iff`: an equivalent definition of the conformality at some point
+* `conformal_at_preserves_angle`: if a map is `conformal_at` some point, then its differential
+                                  preserves all angles at that point
+* `conformal_groupoid`: the groupoid of conformal local homeomorphisms
+
+## Tags
+
+conformal
+
+## Warning
+
+The definition of conformality in this file does NOT require the maps to be orientation-preserving.
+Maps such as the conjugate function on the complex plane is considered as conformal maps.
+-/
+
+noncomputable theory
+
+section conformality
+
+/-- A continuous linear map `f'` is said to be conformal if it's
+a nonzero multiple of a bijective linear isometry. -/
+def is_conformal_map {X Y : Type*} [inner_product_space ℝ X] [inner_product_space ℝ Y]
+  (f' : X →L[ℝ] Y) :=
+∃ (c : ℝ) (hc : c ≠ 0) (lie : X ≃ₗᵢ[ℝ] Y), (f' : X → Y) = c • lie
+
+/-- A map `f` is said to be conformal if it has a conformal differential `f'`. -/
+def conformal_at {X Y : Type*} [inner_product_space ℝ X] [inner_product_space ℝ Y]
+  (f : X → Y) (x : X) :=
+∃ (f' : X →L[ℝ] Y), has_fderiv_at f f' x ∧ is_conformal_map f'
+
+
+namespace conformal_at
+
+variables {X Y Z : Type*}
+  [inner_product_space ℝ X] [inner_product_space ℝ Y] [inner_product_space ℝ Z]
+
+open linear_isometry_equiv continuous_linear_map
+
+theorem differentiable_at {f : X → Y} {x : X} (h : conformal_at f x) :
+  differentiable_at ℝ f x :=
+let ⟨_, h₁, _, _, _, _⟩ := h in h₁.differentiable_at
+
+theorem id (x : X) : conformal_at id x :=
+⟨id ℝ X, has_fderiv_at_id _, 1, one_ne_zero, refl ℝ X, by ext; simp⟩
+
+theorem const_smul {c : ℝ} (h : c ≠ 0) (x : X) : conformal_at (λ (x': X), c • x') x :=
+⟨c • continuous_linear_map.id ℝ X, has_fderiv_at.const_smul (has_fderiv_at_id x) c,
+  c, h, refl ℝ X, by ext; simp⟩
+
+theorem comp {f : X → Y} {g : Y → Z} {x : X} :
+  conformal_at f x → conformal_at g (f x) → conformal_at (g ∘ f) x :=
+begin
+  rintro ⟨f', hf₁, cf, hcf, lief, hf₂⟩ ⟨g', hg₁, cg, hcg, lieg, hg₂⟩,
+  exact ⟨g'.comp f', has_fderiv_at.comp x hg₁ hf₁, cg * cf, mul_ne_zero hcg hcf, lief.trans lieg,
+    by { ext, simp [coe_comp' f' g', hf₂, hg₂, smul_smul cg cf] }⟩,
+end
+
+/-- A real differentiable map `f` is conformal at point `x` if and only if
+    its differential `f'` at that point scales any inner product by a positive scalar. -/
+theorem _root_.conformal_at_iff {f : X → Y} {x : X} {f' : X ≃L[ℝ] Y}
+  (h : has_fderiv_at f f'.to_continuous_linear_map x) :
+  conformal_at f x ↔ ∃ (c : ℝ) (hc : 0 < c),
+  ∀ (u v : X), inner (f' u) (f' v) = (c : ℝ) * (inner u v) :=
```

Comment:
Can you please state the linear-algebra theorem (about `is_conformal_map`) earlier in the file, then deduce the calculus version from it?

Reply:
Also, can you please use the notation `⟪·, ·⟫` for the inner product?  You may need to `open_locale real_inner_product_space` or something like that.